### PR TITLE
Set drop hover width to 50vw

### LINKED ISSUE #3369: CSS-based drop hover width
This issue was created to see if a CSS-based solution is possible to calculate the drop hover width. `useDropHoverWidth` is not the most elegant, as it requires a back channel to set the content width which gets used in the drop hover calculations.

_Responding to @ethan-james in https://github.com/cybersemics/em/issues/3367#issuecomment-3518944345_

> Yeah, unfortunately, the content width doesn't have much of a connection to the width of thoughts, which are absolutely-positioned and some variant of 100% width.

I think that's okay, because the drop hover width is relative to the screen width, breakpoints, and depth of the hovering path, not the editable width.

> Especially when we consider the [960px breakpoint](https://github.com/ethan-james/em/blob/461ad3ad9dde43af4be5eccefc0f4dff2cf052f6/src/components/Content.tsx#L119) where `margin: 0 auto` starts to leave variable-sized gaps on either side of the `Content` component.

That should be pretty easy to calculate in CSS as either a percentage or in `vw`. The drop hover width can also have different breakpoints if different calculations are needed to correspond to different content widths at different screen sizes.

> If the goal is to keep the drop targets the same width as the content pane, then the only CSS-based solution that I can think of involves storing a `--content-width` CSS variable on resize, which is similar in terms of complexity to the existing solution.

The goal is for the drop hover with the deepest path to extend to the right edge of the screen, and then for shallower paths to be shortened by the difference in indentation. For example, if thoughts with depths 2–5 are visible, the drop hover widths might be `100% - padding - depth * indentSize`.

That's off the top of my head. Would need some battle testing to see how this could be implemented in CSS, and if there are other edge cases I'm missing.

### PR DESCRIPTION
**Pull request by ethan-james**
Fixes #3369

In order to simplify drop hover logic, `useDropHoverWidth` now unconditionally returns `50vw` so that we can see if that looks good.

### CODE REVIEW

**Review by raineorshine — CHANGES_REQUESTED** [review]
Just noticing this, but since the value is now constant at compile-time, we should be able to move it into a Panda `css` or `cx` block. It would be good to prefer the Panda approach when possible and reserve `style` for dynamic runtime values. Is that a change you could make before we merge?

File: src/recipes/dropHover.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ const dropHoverRecipe = defineRecipe({
     position: 'absolute',
     height: '3px',
     borderRadius: '99px',
+    width: `token(sizes.dropHover)`,
```
Comment (raineorshine) [inline src/recipes/dropHover.ts]: Simpler:

```
  width: 'dropHover'
```

Comment (ethan-james) [reply src/recipes/dropHover.ts]: It works! Thanks for that.
